ICSharpCode.Decompiler/IL/Transforms/NullPropagationTransform.cs

```diff
@@ -305,11 +340,161 @@ private void IntroduceUnwrap(ILVariable testedVar, ILInstruction varLoad, Mode m
 						refInput: true
 					).WithILRange(varLoad);
 					break;
+				case Mode.UnconstrainedType:
+					// Wrap varLoad in nullable.unwrap:
+					replacement = new NullableUnwrap(varLoad.ResultType, varLoad, refInput: varLoad.ResultType == StackType.Ref);
+					break;
 				default:
 					throw new ArgumentOutOfRangeException(nameof(mode));
 			}
 			oldParentChildren[oldChildIndex] = replacement;
 		}
+
+		// stloc valueTemporary(valueExpression)
```

Comment:
Misleading name: this actually is a stack slot holding the address of the expression before `?`.